### PR TITLE
(CDAP-16676) Upgrade dataproc API v1->v1beta2 and add option to enable component gateway

### PR DESCRIPTION
**Pull request by wyzhang**
Why:
One of the new features in v1beta2 API is allowing enabling dataproc
component gateway which allowing accesses to hadoop web UI, thereby
faciliating debugging.

What:
- Upgrade to use v1beta2 instead of v1
- Add an option to enable component gateway, default is off.

Usage:
Adding the following to cdap-site.xml

provisioner.system.properties.gcp-dataproc.componentGatewayEnabled: "true"

### CODE REVIEW

File: cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocConf.java [hunk 1]
```diff
@@ -378,6 +385,8 @@ static DataprocConf create(Map<String, String> properties, @Nullable SSHPublicKe
                                                                                      "true"));
     boolean stackdriverMonitoringEnabled = Boolean.parseBoolean(properties.getOrDefault(STACKDRIVER_MONITORING_ENABLED,
                                                                                         "true"));
+    boolean componentGatewayEnabled = Boolean.parseBoolean(properties.getOrDefault(COMPONENT_GATEWAY_ENABLED,
```
Comment (chtyim) [inline cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocConf.java]: No need to call `getOrDefault`. just call `get` is ok. `Boolean.parseBoolean` will return `false` for `null`.

**Review by chtyim — APPROVED** [review]
Just one comment.

**Comment by chtyim** [comment]
I think the configuration should be exposed to the UI too. Meaning you'll need to add it to the gcp-dataproc.json under the resources directory.

**Review by albertshau — COMMENTED** [review]
There is a gcp-dataproc.json file that should also be modified, to add this property as a 'toggle' widget in the 'Advanced' section.

See https://github.com/data-integrations/google-cloud/blob/develop/widgets/BigQueryTable-batchsink.json#L116 for an example of the toggle widget

File: cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocConf.java (hunk 1 above)
Comment (wyzhang) [reply cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocConf.java]: Done.

**Comment by wyzhang** [comment]
Done. 

Added to the UI.

File: cdap-runtime-ext-dataproc/src/main/resources/gcp-dataproc.json [hunk 2]
```diff
@@ -162,6 +162,20 @@
             "size": "medium"
           }
         },
+        {
+          "widget-type": "select",
```
Comment (chtyim) [inline cdap-runtime-ext-dataproc/src/main/resources/gcp-dataproc.json]: For true/false, better to use `toggle` type. See https://github.com/cdapio/hydrator-plugins/blob/develop/core-plugins/widgets/File-batchsource.json#L52 for example.

Comment (wyzhang) [reply cdap-runtime-ext-dataproc/src/main/resources/gcp-dataproc.json]: Done.

**Review by chtyim — APPROVED** [review]

**Pull request merged**